### DISCUSSION TITLE
Combined TSP-VRP problem, possible solution?

### DISCUSSION THREAD

**Discussion by albertsallesmosaic** [TimefoldAI/timefold-solver#34] [General]
Hello, first of all, very nice project. In the past I had a close look at OptaPlanner and it's amazing what has emerged from that.

Now, I have a question if it is possible/suitable to use Timefold for my problem. In general, it is a simple VRP with time windows and capacity. There are shipments that start at a PickUp location and end at a Depot location. However, there are intermediate stops, so one shipment goes like this:
PickUp -> Location1 -> ... -> LocationN -> Depot, although in most cases N < 5, so that's not a big deal.

And of course there are multiple shipments that start/end at different locations and a fleet of trucks. Each truck can only carry one shipment at a time.

My question really is whether this is just two problems combined:
1. Finding the best path from PickUp to Depot as a TSP for each shipment.
2. A VRP with time windows where each task (shipment) starts and ends at different locations.

If you could give me a hand on how to tackle this problem with Timefold (and if it's suitable). I've been going through the docs but I cannot seem to find how to do this. Thanks!!

(Sorry I didn't see the discussions tab and I created an issue by mistake)

**Comment by triceo** [comment]
Overtime, we will be extending the set of quickstarts to include variants of VRP such as Time Windows and Pickup&Delivery. For now though, we do not have a ready-made example to share.
